presto-hive/src/test/java/com/facebook/presto/hive/TestHiveIntegrationSmokeTest.java

```diff
@@ -2824,22 +2826,59 @@ public void testInsertMultipleColumnsFromSameChannel()
     public void testCreateAvroTableWithSchemaUrl()
             throws Exception
     {
-        File tempDir = createTempDir();
-        File schemaFile = createSchemaFileIn(tempDir);
-        String createTableSql = getAvroCreateTableSql(schemaFile.getAbsolutePath());
-        String expectedSql = getAvroCreateTableSql(schemaFile.toURI().toString());
+        String tableName = "test_create_avro_table_with_schema_url";
+        File schemaFile = createAvroSchemaFile();
+
+        String createTableSql = getAvroCreateTableSql(tableName, schemaFile.getAbsolutePath());
+        String expectedShowCreateTable = getAvroCreateTableSql(tableName, schemaFile.toURI().toString());
 
         assertUpdate(createTableSql);
-        MaterializedResult actual = computeActual("SHOW CREATE TABLE test_create_avro");
-        assertEquals(actual.getOnlyValue(), expectedSql);
 
-        assertUpdate("DROP TABLE test_create_avro");
-        deleteRecursively(tempDir.toPath(), ALLOW_INSECURE);
+        try {
+            MaterializedResult actual = computeActual(format("SHOW CREATE TABLE %s", tableName));
+            assertEquals(actual.getOnlyValue(), expectedShowCreateTable);
+        }
+        finally {
+            assertUpdate(format("DROP TABLE %s", tableName));
+            verify(schemaFile.delete(), "cannot delete temporary file: %s", schemaFile);
+        }
     }
 
-    private String getAvroCreateTableSql(String schemaFile)
+    @Test
+    public void testAlterAvroTableWithSchemaUrl()
+            throws Exception
     {
-        return format("CREATE TABLE %s.%s.test_create_avro (\n" +
+        testAlterAvroTableWithSchemaUrl(true, true, true);
+    }
+
+    protected void testAlterAvroTableWithSchemaUrl(boolean renameColumn, boolean addColumn, boolean dropColumn)
+            throws Exception
+    {
+        String tableName = "test_alter_avro_table_with_schema_url";
+        File schemaFile = createAvroSchemaFile();
+
+        assertUpdate(getAvroCreateTableSql(tableName, schemaFile.getAbsolutePath()));
+
+        try {
+            if (renameColumn) {
+                assertQueryFails(format("ALTER TABLE %s RENAME COLUMN dummy_col TO new_dummy_col", tableName), "ALTER TABLE not supported when Avro schema url is set");
+            }
+            if (addColumn) {
+                assertQueryFails(format("ALTER TABLE %s ADD COLUMN new_dummy_col VARCHAR", tableName), "ALTER TABLE not supported when Avro schema url is set");
+            }
+            if (dropColumn) {
+                assertQueryFails(format("ALTER TABLE %s DROP COLUMN dummy_col", tableName), "ALTER TABLE not supported when Avro schema url is set");
+            }
+        }
+        finally {
+            assertUpdate(format("DROP TABLE %s", tableName));
```

Comment:
you could introduce a `TemporaryAvroTable` that could be `Closeable`. Then it would be it would be less copy-paste and tests would be cleaner.

Reply:
We had something similar in other places. That make sense. However if we decided to go that way, we need to change the other tests as well. I'm going to keep it as is for now.